validator_worker/src/follower.rs

```diff
@@ -333,3 +333,527 @@ async fn on_error<'a, C: Unlocked + 'static>(
+                    channel_context.context.id(),
+                    channel_context.token.precision.get(),
+                )
+                .expect("should encode");
+            let new_state: NewState<UncheckedState> = NewState {
+                state_root: state_root.clone(),
+                signature: IDS[&*LEADER].to_checksum(),
+                balances: get_initial_balances().into_unchecked(),
+            };
+            let _mock_guard = server_setup.setup_new_state_response(Some(new_state)).await;
+
+            let reject_state = RejectState {
+                state_root,
+                signature: IDS[&*FOLLOWER].to_checksum(),
+                timestamp: Some(received),
+                reason: "rejected".to_string(),
+                balances: None,
+            };
+            let _mock_guard = server_setup
+                .setup_reject_state_response(Some(reject_state))
+                .await;
+            let tick_status = tick(
+                &sentry,
+                &channel_context,
+                get_initial_spenders(),
+                get_initial_balances(),
+            )
+            .await
+            .expect("Shouldn't return an error");
+            assert!(matches!(
+                tick_status.approve_state,
+                ApproveStateResult::Sent(None)
+            ));
+        }
+
+        // - Case where output will be ApproveStateResult::Sent(Some(propagation_result)) (all rules have been met)
+        {
+            let balances = get_initial_balances();
+            let state_root = balances
+                .encode(
+                    channel_context.context.id(),
+                    channel_context.token.precision.get(),
+                )
+                .expect("Should encode");
+            let _mock_guard = server_setup
+                .setup_last_approved_response(balances.into_unchecked(), state_root)
+                .await;
+
+            let state_root = get_initial_balances()
+                .encode(
+                    channel_context.context.id(),
+                    channel_context.token.precision.get(),
+                )
+                .expect("should encode");
+            let new_state: NewState<UncheckedState> = NewState {
+                state_root: state_root,
+                signature: IDS[&*LEADER].to_checksum(),
+                balances: get_initial_balances().into_unchecked(),
+            };
+            let res = on_new_state(
+                &sentry,
+                &channel_context,
+                get_initial_balances(),
+                new_state,
+                UnifiedNum::from_u64(1_000_000),
+            )
+            .await
+            .expect("Shouldn't return an error");
+            assert!(matches!(res, ApproveStateResult::Sent(Some(..))));
+            let propagated_to = match res {
+                ApproveStateResult::Sent(propagated_to) => propagated_to,
+                _ => None, // Shouldn't happen
+            };
```

Comment:
You can just match against:
`ApproveStateResult::Sent(Some(propagated_to))` and `panic!`